src/Database/RavenCommand.ts

```diff
@@ -86,7 +86,7 @@ export abstract class RavenCommand {
     return options;
   }
 
-  public setResponse(response: IResponse): IRavenCommandResponse | null | void {
+  public setResponse(response: IResponse): IRavenResponse | IRavenResponse[] | null | void {
```

Comment:
For the sake of `BatchCommand` we're changing `RavenCommand` to accept object or array of objects. In C# RavenCommand is generic and BatchCommand extends it like this:

```
    public class BatchCommand : RavenCommand<BlittableArrayResult>, IDisposable
    {
...
```

Could we do it like this? `IRavenResponse | IRavenResponse[] | null | void` as a return type is really confusing. It would be great if we could just leave IRavenResponse.